lumen/sources/snowflake.py

```diff
@@ -117,3 +117,87 @@ def get(self, table, **query):
         for st in sql_transforms:
             sql_expr = st.apply(sql_expr)
         return self.execute(sql_expr)
+
+    @cached
+    def get_table_metadata(self, table_name: str) -> dict[str, dict]:
+        """
+        Generate metadata for a single table in Snowflake.
+
+        Args:
+            table_name: Name of the table to get metadata for
+
+        Returns:
+            A dictionary with table metadata in the format:
+            {"description": ..., "columns": {"column_name": "description", ...}}
+        """
+        # Get table description
+        table_query = f"""
+            SELECT
+                TABLE_NAME,
+                COMMENT as TABLE_DESCRIPTION
+            FROM
+                {self.database}.INFORMATION_SCHEMA.TABLES
+            WHERE
+                TABLE_NAME = '{table_name}'
+        """
+
+        table_metadata = self.execute(table_query)
+
+        if table_metadata.empty:
+            return {"description": "", "columns": {}}
+
+        description = table_metadata.iloc[0]['TABLE_DESCRIPTION']
+        result = {
+            "description": description if description else "",
+            "columns": {}
+        }
+
+        # Get column metadata
+        column_query = f"""
+            SELECT
+                COLUMN_NAME,
+                COMMENT
+            FROM
+                {self.database}.INFORMATION_SCHEMA.COLUMNS
+            WHERE
+                TABLE_NAME = '{table_name}'
+            ORDER BY
+                ORDINAL_POSITION
+        """
+
+        columns_info = self.execute(column_query)
+
+        for _, row in columns_info.iterrows():
+            column_name = row['COLUMN_NAME']
+            description = row['COMMENT']
+            result["columns"][column_name] = description if description else ""
+
+        return result
+
+    def get_metadata(self, tables: dict[str, str] | None = None) -> dict[str, dict]:
```

Comment:
I know it's somewhat overloaded but I'd follow the pattern we used for `.get` and `.get_schema` and have one method that accepts a `table` argument, which, when undefined, returns data for all tables.

Minor comments:
- Type declaration seems wrong and docstring is wrong format.